test/CMakeLists.txt

```diff
@@ -67,6 +70,18 @@ macro(onedpl_add_test test_source_file)
         target_compile_options(${_test_name} PRIVATE /bigobj)
     endif()
 
+    # Disable checked iterators on Windows for debug mode
+    # For details please see
+    #    https://learn.microsoft.com/en-us/cpp/standard-library/iterator-debug-level?view=msvc-170
+    #    https://learn.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-170
+    #    https://stackoverflow.com/questions/51494506/replacing-md-with-mt-in-cmake-is-not-possible-in-release-mode
+    if (WIN32 AND DEFINED ${_build_type_in_lower} AND ${_build_type_in_lower} STREQUAL "debug")
+        if (${switch_off_checked_iterators})
```

Comment:
Just a small suggestion: the condition for `${switch_off_checked_iterators}` can be included in the first `if`. That way, you can have a single `if` statement instead of nested statements.